src/pages/Navgurukul/merakiAdmission/index.js

```diff
@@ -0,0 +1,211 @@
+import React, { useState, useEffect } from "react";
+import axios from "axios";
+import { toast } from "react-toastify";
+import YouTube from "react-youtube";
+import { METHODS } from "../../../services/api";
+import "./styles.scss";
+
+function Admission() {
+  const [userDetails, setUserDetails] = useState({
+    firstName: "",
+    middleName: "",
+    lastName: "",
+    mobileNumber: "",
+  });
+
+  const [mobile, setMobile] = useState("");
+  const [enrolmentKey, setEnrolmentKey] = useState("");
+
+  const [partnerId, setPartnerId] = useState("");
+
+  let userToken = localStorage.getItem("Token");
+  useEffect(() => {
+    axios({
+      method: METHODS.GET,
+      url: `${process.env.REACT_APP_MERAKI_URL}/users/me`,
+      headers: {
+        accept: "application/json",
+        Authorization: userToken,
+      },
+    }).then((res) => {
+      setPartnerId(res.data.user.partner_id);
+    });
+  }, []);
+
+  const testUrl = "http://dev-join.navgurukul.org/k/";
+
+  const generateTestLink = async () => {
+    try {
+      const mobile = `0${userDetails.mobileNumber}`;
+      const dataURL = `${process.env.REACT_APP_CHANAKYA_BASE_URL}helpline/register_exotel_call`;
+      const response = await axios.get(dataURL, {
+        params: {
+          ngCallType: "getEnrolmentKey",
+          From: mobile,
+          partner_id: partnerId ? partnerId : null,
+        },
+      });
+      setEnrolmentKey(response.data.key);
+      const params = {
+        firstName: userDetails.firstName,
+        middleName: userDetails.middleName,
+        lastName: userDetails.lastName,
+        mobileNumber: userDetails.mobileNumber,
+      };
+      const queryString = Object.keys(params)
+        .map((filter) => {
+          if (params[filter]) {
+            return `${filter}=${params[filter]}`;
+          }
+          return null;
+        })
+        .filter((item) => item)
+        .join("&");
+
+      const url = `${testUrl}${enrolmentKey}?${queryString}`;
+
+      window.open(url, "_blank");
+      return response;
+    } catch (e) {}
+  };
+
+  const giveTest = async () => {
+    if (
+      !userDetails.firstName ||
+      !userDetails.lastName ||
+      !userDetails.mobileNumber
+    ) {
+      toast.error(
+        "To attempt the test, it is compulsory to enter your First Name, Last Name and Mobile Number. Middle Name is optional, you can choose not to enter.",
+        {
+          position: toast.POSITION.BOTTOM_RIGHT,
+        }
+      );
+      return;
+    }
+    if (userDetails.mobileNumber.toString().length !== 10) {
```

Comment:
You can use a [regular expression test](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test) to make sure that the string only consists of (exactly 10) digits:
```suggestion
    if (/[0-9]{10}/.test(userDetails.mobileNumber.toString())) {
```

Reply:
If I do like this with `[0-9]{10}` then it won't throw an error when the length is greater than 10